sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveMetadataCacheSuite.scala

```diff
@@ -91,6 +95,19 @@ class HiveMetadataCacheSuite extends QueryTest with SQLTestUtils with TestHiveSi
             // Test refreshing the cache.
             spark.catalog.refreshTable("test")
             assert(sql("select * from test").count() == 4)
+            assert(spark.table("test").inputFiles.length == 4)
+
+            // Test refresh by path separately since it goes through different code paths than
+            // refreshTable does.
+            deleteRandomFile()
+            spark.catalog.cacheTable("test")
+            spark.catalog.refreshByPath("/some-invalid-path")  // no-op
+            val e2 = intercept[SparkException] {
+              sql("select * from test").count()
+            }
+            assert(e2.getMessage.contains("FileNotFoundException"))
+            spark.catalog.refreshByPath(dir.getAbsolutePath)
```

Comment:
Note: before #14690, users need to refresh one of the partition paths to invalide the cache, but now they have to refresh the table path, because `TableFileCatalog.rootPaths` only contains table path while `ListingFileCatalog.rootPaths` only contains partition paths.

I think it's better than before, but it's still a breaking change, should we docuement it in the 2.1 release notes?